tests/console/sshd.pm

```diff
@@ -126,16 +129,58 @@ sub run {
     assert_script_run "scp -4v '$ssh_testman\@localhost:/etc/{group,passwd}' /tmp";
     assert_script_run "scp -4v '$ssh_testman\@localhost:/etc/ssh/*.pub' /tmp";
 
+    # poo#80716 Test all available ciphers, key exchange algorithms, host key algorithms and mac algorithms.
+    # TODO: This does not work for Tumbleweed because of nmap
+    # See pull request #11930 for more details
+    test_cryptographic_policies(remote_user => $ssh_testman) unless (is_opensuse);
+
+    assert_script_run "killall -u $ssh_testman || true";
+    wait_still_screen 3;
+
     # Restore ~/.ssh generated in consotest_setup
     # poo#68200. Confirm the ~/.ssh_bck directory is exist in advance and then restore, in order to avoid the null restore
     assert_script_run 'rm -rf ~/.ssh';
     assert_script_run 'if [ -d ~/.ssh_bck ]; then mv ~/.ssh_bck ~/.ssh; fi';
 
-    assert_script_run "killall -u $ssh_testman || true";
-    wait_still_screen 3;
+    # Restore the /etc/ssh/sshd_config
+    assert_script_run 'cp /etc/ssh/sshd_config{_before,}';
+
+    record_info("Restart sshd", "Restart sshd.service");
+    systemctl("restart sshd");
+
     clear_console if !is_serial_terminal;
 }
 
+sub test_cryptographic_policies() {
+    my %args        = @_;
+    my $remote_user = $args{remote_user};
+
+    my @crypto_params = (["Ciphers", "cipher", "-c "], ["HostKeyAlgorithms", "key", "-o HostKeyAlgorithms="], ["KexAlgorithms", "kex", "-o kexalgorithms="], ["MACS", "mac", "-m "]);
```

Comment:
```suggestion
    my @crypto_params = (["Ciphers", "cipher", "-c "], KexAlgorithms", "kex", "-o kexalgorithms="], ["MACS", "mac", "-m "]);
    push(@crypto_params, ["HostKeyAlgorithms", "key", "-o HostKeyAlgorithms="]) unless (is_opensuse);
```

Reply:
This would ensure that the rest of the algorithms are still checked, apart from the host_key algorithms, even in opensuse.

Reply:
This is great! Thank you 👍